r/tests/testthat/test-dplyr-join.R

```diff
@@ -249,3 +249,24 @@ test_that("arrow dplyr query correctly filters then joins", {
     )
   )
 })
+
+
+test_that("arrow dplyr query can join with tibble", {
+  # ARROW-14908
+  existing_use_threads <- getOption("arrow.use_threads")
+  options(arrow.use_threads = FALSE)
```

Comment:
Nice test! This might be better to use `withr::with_options(list(arrow.use_threads = FALSE), { ... })` and then you don't need to worry about resetting later

Similar to https://github.com/apache/arrow/blob/3b9462a4ffc9f1d20ffc4ba578adec0f0ed8ffbd/r/tests/testthat/test-parquet.R#L302-L313